soc/xtensa/esp32/Kconfig.defconfig

```diff
@@ -18,4 +18,7 @@ config XTENSA_ASM2
 config MP_NUM_CPUS
 	default 2
 
+config NEWLIB_LIBC_AVAILABLE
```

Comment:
this is a toolchain feature, not related to SoC, if we are introducing this just to filter in tests, I suggest to export this in toolchain files and filter on those exports instead of a kconfig and make it SoC specific.

Reply:
That sounds like a lot of spots to find to add a "ZEPHYR_TOOLCHAIN_HAS_NEWLIB" variable though.

My other thought was to augment cmake to probe for newlib configure-style, but that sounded complicated.

Would you be OK with just reverting this trick and simply blacklisting the tests individually the way it's done for esp32 already, then fixing it out of band in a separate bug?

Reply:
let me try and propose a solution based on SDK features.

Reply:
i tried to address this with #12393

Reply:
> Would you be OK with just reverting this trick and simply blacklisting the tests individually the way it's done for esp32 already, then fixing it out of band in a separate bug

Please let's do that, #12393 has turned out to be controversial and I'd like to see this patch merged soon